src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -291,7 +293,9 @@ bool ThetaStarPathPlanner::findPathToEnd(const Coordinate &end_coord)
         // Add this vertex to the closed list
         closed_list.insert(current_coord);
 
-        if (visitNeighbours(current_coord, end_coord))
+        // Check if current coordinate or its surrounding neighbours are at the
+        // destination
+        if (current_coord == end_coord || visitNeighbours(current_coord, end_coord))
```

Comment:
is this change still necessary?

Reply:
You're correct, it's not needed anymore. Updated